thirdparty/hairer/radau5_c_py.pyx

```diff
@@ -106,21 +97,146 @@ cdef int callback_solout(integer* nrsol, doublereal* xosol, doublereal* xsol, do
     cdef np.ndarray[double, ndim=1, mode="c"]y_py = np.empty(nsolu[0], dtype = np.double)
     cdef np.ndarray[double, ndim=1, mode="c"]cont_py = np.empty(4*nsolu[0], dtype = np.double)
     cdef np.ndarray[double, ndim=1, mode="c"]werr_py = np.empty(nsolu[0], dtype = np.double)
-    c2py(y_py, y, nsolu[0])
-    c2py(cont_py, cont, 4*nsolu[0])
-    c2py(werr_py, werr, nsolu[0])
+    c2py_d(y_py, y, nsolu[0])
+    c2py_d(cont_py, cont, 4*nsolu[0])
+    c2py_d(werr_py, werr, nsolu[0])
 
     irtrn[0] = (<object>solout_PY)(nrsol[0], xosol[0], xsol[0],
                                    y_py, cont_py, werr_py,
                                    lrc[0], irtrn[0])
 
     return irtrn[0]
 
+cdef class RadauSuperLUaux:

```

Comment:
I propose to move this into C instead of Pyx. There is value of having this code there as well (i.e. to simplify using this outside of Assimulo)

I propose to introduce something similar as to what Sundials are using. A create method, in C, which creates these structures, then the regular Radau call to integrate the problem and a finalize method. Example:

Radau5Struct Radau5Create(size_t dim, size_t nnz) { ... }
Radau5Solve(Radau5Struct ...) {...}
Radau5Free(...)

Reply:
Fixed in new commit, this actually simplified some Radau function calls quite a bit. 

I kept this largely separate from the superLU interface for now.

Also, a basic class had to remain on the pyx level to be able to store the data structures over multiple intergrate calls. Unlike CVode, the primary Radau class in on the Python level after all.

Reply:
Ok great that it simplified the code!